### PR TITLE
Update network plugins page for cni spec v1.0.0

### PR DESCRIPTION
**Pull request by reylejano**
This PR updates the network plugins page for CNI spec v1.0.0 as of Kubernetes v1.24

CNI spec bump from 0.4.0 to 1.0.0 in https://github.com/containernetworking/cni/pull/854
Kubernetes tracking issue: kubernetes/kubernetes#109686

### CODE REVIEW

**Comment by reylejano** [comment]
cc: @mikebrow @rolfedh 
/assign @nate-double-u

**Comment by sftim** [comment]
If we are bumping the expected CNI spec version, does that need its own entry in the release notes (maybe a one-line thing)?

That'd be separate from the “No, really, you must read this before you upgrade” item.

**Comment by sftim** [comment]
@reylejano for a change like this, I'd expect to see the PR description mentioning an issue that this addresses. The reason I'm asking for an issue is that we use those to anchor and track any discussion about the change.

**Comment by reylejano** [comment]
Converted to draft
Waiting for the related tracking issue to be created and looking for related PRs or KEP

**Comment by sftim** [comment]
Just so it's clear, I think this is a great idea and will readily approve this change once the context for the change is clear.

**Comment by sftim** [comment]
I'm not sure if we should also recommend downloading a different version of the CNI tools (see https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/).

**Review by rolfedh — APPROVED** [review]

**Comment by aojea** [comment]
the CNI, since the dockershim deprecation, is part of the Container Runtime only  now , there is no code in kubernetes/kubernetes that requires it

**Comment by sftim** [comment]
We're essentially now saying “Kubernetes expects its ecosystem to be using CNI 1.x”, right?

**Comment by squeed** [comment]
> We're essentially now saying “Kubernetes expects its ecosystem to be using CNI 1.x”, right?

FWIW, the CNI protocol is backwards-compatible. So CNI plugins will work with whatever version your runtime supports. So this really isn't a drastic change from a systems administration point of view.

**Comment by sftim** [comment]
OK, so it wouldn't need a KEP but a release note would be appropriate, I'm thinking.

@reylejano would you be willing to make sure that the bump to the expected version makes it into the Kubernetes release notes as its own item (separate from the callout about containerd and CRI-O and how they configure network plugins)? If we have that in the v1.24 release notes then I think this'd be OK to land (subject to SIG Network and SIG Node signing off on it).

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md
Comment (mikebrow) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md:18]: suggest 

```
You must use a CNI plugin that is compatible with the [v0.4.0](https://github.com/containernetworking/cni/blob/spec-v0.4.0/SPEC.md) or later releases of the CNI specification. The current version of CNI is [v1.0.0](https://github.com/containernetworking/cni/blob/spec-v1.0.0/SPEC.md) release of the CNI specification.
```

**Review by mikebrow — COMMENTED** [review]
see comment

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md
Comment (nate-double-u) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md:18]: is `v1.0.0` still correct, or should it be `v1.1.0` now?

Comment (reylejano) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]: Added this suggestion and squashed commits

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md:20]: This could go stale if a new spec version comes out.

Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md:20]: How about this:
```suggestion
You must use a CNI plugin that is compatible with the 
[v0.4.0](https://github.com/containernetworking/cni/blob/spec-v0.4.0/SPEC.md) or later
releases of  the CNI specification. The Kubernetes project recommends using a plugin that is
compatible with the [v1.0.0](https://github.com/containernetworking/cni/blob/spec-v1.0.0/SPEC.md)
CNI specification (plugins can be compatible with multiple spec versions).
```

which avoids the statement that v1.0.0 is current. Any time that we make an assertion about the latest release of an external project, we're adding maintenance for ourselves. I prefer wording that is closer to being [timeless](https://developers.google.com/style/timeless-documentation).

**Comment by sftim** [comment]
@kubernetes/sig-node-pr-reviews highlighting this for review

**Comment by sftim** [comment]
LGTM for SIG Docs

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md
Comment (reylejano) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]: Accepted @sftim's suggestion and squashed commits

**Comment by sftim** [comment]
LGTM for SIG Docs

**Review by mikebrow — APPROVED** [review]
LGTM

**Comment by nate-double-u** [comment]
/approve

Thanks everyone!

**Comment by nate-double-u** [comment]
oops, need an /lgtm too. Adding here as we got bot-unreadable ones from docs and tech :)

/lgtm

**Comment by nate-double-u** [comment]
/hold-remove
since we have the tech review

(next time i should do all three in one comment 😅)

**Pull request merged**
